src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -473,6 +467,8 @@ CHIP_ERROR UDPEndPointImplLwIP::IPv4JoinLeaveMulticastGroupImpl(InterfaceId aInt
     const ip4_addr_t lIPv4Address = aAddress.ToIPv4();
     err_t lStatus;
 
+    ScopedLwIPLock lwipLock;
```

Comment:
Would be better to scope this as narrowly as possible (i.e. not including the `MapErrorLwIP` in the scope).

Reply:
Narrowed the scope.